### DISCUSSION TITLE
add extraModalAction with submit

### DISCUSSION THREAD

**Discussion by panacz** [filamentphp/filament#7218] [Help] [answered]
Hello! I would like to place an extra button in a modal, which performs an additional function on the record in addition to simple saving (switches a boolean value).

At the moment, I have come to the point where this button does the same thing as the base. Where can I enter the extra function?

```
Tables\Actions\Action::make('answer')
    ->action(function (SupportTicket $record, array $data): void {
        $record->sendAdminMessage($data);
        Notification::make()
            ->title('Sikeressen üzenetküldés')
            ->success()
            ->send();
    })
    ->modalContent(fn (SupportTicket $record) => view('filament.pages.actions.messages', ['record' => $record]))
    ->form([
    ])
    ->extraModalActions(fn (SupportTicket $record, array $data): array => [
        Tables\Actions\Action::makeModalAction('answerAndClose')
            ->submit()
            ->color('secondary')
            ->label('Üzenet küldése és zárás')
    ])
    ->modalButton('Üzenet küldése')
```

**Comment by danharrin** [comment]
https://beta.filamentphp.com/docs/3.x/actions/modals#adding-an-extra-modal-action-button-to-the-footer

**Reply by panacz** [reply]
Thanks for the reply! I won't be able to do this in v2?

**Reply by danharrin** [reply, marked as answer]
It was similar in v2, example https://github.com/filamentphp/filament/blob/2.x/packages/tables/src/Actions/CreateAction.php#L35-L40

**Reply by panacz** [reply]
I don't know if this is the best solution, but it worked with 'callMountedTableAction':
```
->extraModalActions(fn (): array => [
    Tables\Actions\Action::makeModalAction('answerAndClose')
        ->action('callMountedTableAction', ['close' => true])
        ->color('secondary')
        ->label('Üzenet küldése és zárás'),
])
```
Thanks for the help!
